### ISSUE TITLE
No animated expansion for Gridview 

### ISSUE THREAD

**Issue by EvgrafovBTW** [flutter/flutter#117602]
I wanted to create a gridview which would look like 4 items (with 2 items cross axis length) and a "show more" button, upon pressing which more items would appear. This works if i just add more entries to the list of aforementioned items but there is no animation to the process. I tried to use the Animated Container widget, but to do that i needed the height of the grid. Long story short I had to install third party package which allows to create grid where elements could have unique height. Thus manually calculating height of the grid i got animation through Animated Container. 

But this is just not a good solution.

I would love to have a Gridview which would expand with animation FROM certain iteration in main axis TO the last iteration. And collapse.

Thank you for your attention. If what I am proposing is already exists please note it out or give away information about where I could find more about the problem.

**Comment by huycozy** [comment]
@EvgrafovBTW Can you share a visual of the expected result for this (or an existing app containing this feature)? 

If there is nothing exists like that, can you share a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that you are trying to implement? Thank you!

**Comment by EvgrafovBTW** [comment]
> If there is nothing exists like that, can you share a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that you are trying to implement? Thank you!

@huycozy, yes, i am not sure how to add a visual, so there is an example project with the animated grid.
[https://github.com/EvgrafovBTW/flutter_expanded_grid_issue.git](url)

In the body of a Scaffold i have an AnimatedContainer which contains [3rd party gridView](https://pub.dev/packages/dynamic_height_grid_view) where i can change _height_ of grid's children (inside Container which is a child here).
`AnimatedContainer(
                duration: const Duration(milliseconds: 500),
                height: seeAll ? fullGridHeight : shortGridHeight,
                child: DynamicHeightGridView(
                    physics: const NeverScrollableScrollPhysics(),
                    shrinkWrap: true,
                    builder: ((context, index) {
                      return productCards[index];
                    }),
                    crossAxisSpacing: 15,
                    mainAxisSpacing: 25,
                    itemCount: productCards.length,
                    crossAxisCount: 2),
              ),
`
This allows me to calculate height of the AnimatedContiner and change it _onTap:_
`GestureDetector(
                onTap: () {
                  setState(() {
                    seeAll = !seeAll;
                  });
                },`
` gridColumnLength = (productCards.length / 2).roundToDouble();
    fullGridHeight = gridColumnLength * 200 + gridColumnLength * 25;
    shortGridHeight = shorGridItems * 200 + shorGridItems * 25;`

But this is just not right. Would be great if we could have _GridView_ which would support animation between its full\collapsed state like _AnimatedContainer_, but instead of pixel height it would use provided by user **index** of a bottom (last) row in a**collapsed** state.

**Comment by huycozy** [comment]
@EvgrafovBTW thanks for providing the sample code. Adding the demo as below as a visual sample for this.

<details>
<summary>Demo</summary>

| GridView | DynamicHeightGridView |
| --------------- | --------------- |
<video src="https://user-images.githubusercontent.com/104349824/209613335-fcb2d735-f245-40cb-9a50-9c3c64373848.mov"/> | <video src="https://user-images.githubusercontent.com/104349824/209613239-5a42cedc-4305-4653-921e-a9f0686b87e6.mov"/>

</details>

I'm not sure if this feature will be supported or not, so I'm labeling the issue as a proposal for further insights from the team.

**Comment by EvgrafovBTW** [comment]
@huycozy Thanks!
